service_configuration_lib/spark_config.py

```diff
@@ -250,6 +250,35 @@ def _append_sql_shuffle_partitions_conf(spark_opts: Dict[str, str]) -> Dict[str,
     return spark_opts
 
 
+def _get_dra_configs(spark_opts: Dict[str, str]) -> Dict[str, str]:
+    if ('spark.dynamicAllocation.enabled' not in spark_opts or
+        str(spark_opts['spark.dynamicAllocation.enabled']) != 'true'):
+        return spark_opts
+
+    spark_opts['spark.dynamicAllocation.shuffleTracking.enabled'] = "true"
+    spark_opts['spark.dynamicAllocation.executorAllocationRatio'] = "0.8"
+
+    if 'spark.dynamicAllocation.minExecutors' not in spark_opts:
+
+        min_executors = int(spark_opts.get('spark.executor.instances', DEFAULT_EXECUTOR_INSTANCES)) // 4
+        if 'spark.dynamicAllocation.initialExecutors' in spark_opts:
+            min_executors = min(min_executors, int(spark_opts.get('spark.dynamicAllocation.initialExecutors')))
+
+        spark_opts['spark.dynamicAllocation.minExecutors'] = str(min_executors)
```

Comment:
Add a warning message, if minExecutors was specified but overwritten with initialExecutors.
Though I believe spark does that implicitly? If so, I would suggest ignore the overriding.

Reply:
I am not overriding minExecutors if it was specified. I'm only setting it to the minimum of `spark.executor.instances // 4` and `spark.dynamicAllocation.initialExecutors` if minExecutors is not specified already